### PR TITLE
172 index doc

### LINKED ISSUE #172: Document or improve utils/serializers/index.ts

### PR DESCRIPTION
**Pull request by windushka**
aims to resolve #172

### CODE REVIEW

File: src/utils/serializers/index.ts [hunk 1]
```diff
@@ -1,4 +1,25 @@
+/**
+ * This module exports functions for working with strings, including encoding and decoding
+ * between string and binary representations, such as Uint8Array.
+ */
```
Comment (Elli610) [inline src/utils/serializers/index.ts:4]:
```suggestion
/**
 * This module exports encoding and decoding functions for strings.
 */
```
we should keep it simple

File: src/utils/serializers/index.ts [hunk 2]
```diff
@@ -1,4 +1,25 @@
...
 export * from './strings';
+
+/**
+ * This module exports functions for working with numeric data types, such as integers and
+ * floating-point numbers, and their binary representations in Uint8Array. The functions
+ * handle signed and unsigned integers, as well as BigInt for 64-bit integers.
```
Comment (Elli610) [inline src/utils/serializers/index.ts:10]:
```suggestion
 * This module exports functions needed to process numeric data, such as integers and 
 * floating point numbers and their binary representations in Uint8Array. 
 * The functions handle signed and unsigned integers, as well as BigInt for 64-bit integers.
```

File: src/utils/serializers/index.ts [hunk 3]
```diff
@@ -1,4 +1,25 @@
...
+/**
+ * This module exports functions for working with boolean values, including functions for
+ * converting between boolean values and their binary representation in Uint8Array.
+ */
 export * from './bool';
+
+/**
+ * This module exports functions for working with arrays, including functions for manipulating
+ * and transforming arrays of various data types, such as integers, floating-point numbers,
+ * and other nested arrays.
```
Comment (Elli610) [inline src/utils/serializers/index.ts:23]:
```suggestion
 * This module exports serialization and deserialization functions for arrays containing native types.
```
same here

**Review by Elli610 — CHANGES_REQUESTED** [review]
changes needed

**Review by Elli610 — CHANGES_REQUESTED** [review]
The @throws should be located between @param and @returns, check notion

File: src/utils/serializers/numbers.ts [hunk 4]
```diff
@@ -21,18 +30,28 @@ export function u8toByte(val: number): Uint8Array {
 }
 
 /**
- * Converts a Uint8Array into a u8.
+ * Converts a Uint8Array into an unsigned 8-bit integer (u8).
  *
- * @param arr - the array to convert
+ * @param arr - The array to convert
+ *
```
Comment (Elli610) [inline src/utils/serializers/numbers.ts:36]:
```suggestion
```

**Review by Elli610 — APPROVED** [review]

**Review by 0xSwapFeeder — APPROVED** [review]

**Review by Ben-Rey — APPROVED** [review]

**Pull request merged**